python/ray/experimental/gpu_object_manager/gpu_object_manager.py

```diff
@@ -20,19 +38,55 @@ class GPUObjectMeta(NamedTuple):
     # Must be a valid backend name as defined in
     # `ray.util.collective.types.Backend`.
     tensor_transport_backend: str
-    tensor_meta: List[Tuple["torch.Size", "torch.dtype"]]
+    tensor_meta: TensorMetadata
 
 
-def __ray_get_tensor_meta__(self, obj_id: str):
-    """Helper function that runs on the src actor to get the tensor metadata."""
+def __ray_get_tensor_meta__(
+    self: "ray.actor.ActorHandle", obj_id: str, use_nixl: bool
+) -> TensorMetadata:
+    """Helper function that runs on the source actor to get tensor metadata.
+
+    This function retrieves metadata about tensors stored in the GPU object store,
+    including their shapes and dtypes. When NIXL transport is enabled, it also gets
+    NIXL-specific metadata needed for tensor transport.
+
+    Args:
+        self: The actor that runs this function.
+        obj_id: The ID of the GPU object to get metadata for
+        use_nixl: Whether to use NIXL transport and get NIXL metadata
+
+    Returns:
+        TensorMetadata: A named tuple containing the tensor metadata.
+        - tensor_basic_meta: A list of tuples, each containing the shape and dtype of a tensor.
+        - nixl_serialized_descs: Serialized NIXL descriptors for NIXL transport.
+        - nixl_agent_meta: The additional metadata of the remote NIXL agent.
+
+    Raises:
+        AssertionError: If the object ID is not found in the GPU object store
+    """
     from ray._private.worker import global_worker
 
     gpu_object_store = global_worker.gpu_object_manager.gpu_object_store
     assert gpu_object_store.has_gpu_object(
         obj_id
     ), f"obj_id={obj_id} not found in GPU object store"
     tensors = gpu_object_store.get_gpu_object(obj_id)
-    return [(t.shape, t.dtype) for t in tensors]
+    if use_nixl:
+        from ray.util.collective.collective_group.nixl_backend import NixlBackend
+
+        nixl_backend: NixlBackend = collective.get_group_handle("nixl")
```

Comment:
It's better not to assume that NIXL's group name is `nixl`. How about passing the group into this function and then retrieving its group handle's backend to check whether it's NIXL or not?